### PR TITLE
[move] remove unnecessary extra import of move

### PR DESCRIPTION
**Pull request by davidiw**
### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2616)

### CODE REVIEW

File: aptos-move/aptos-gas/Cargo.toml [hunk 1]
```diff
@@ -10,12 +10,5 @@ publish = false
 edition = "2021"
 
 [dependencies]
-move-binary-format = { git = "https://github.com/move-language/move", rev = "e97dc204943776e80a16255432ee4e92c4e08652" }
```
Comment (wrwg) [inline aptos-move/aptos-gas/Cargo.toml:13]: This was discussed on the channel, for now it is not unnecessary, somehow related to feature unification.

**Review by vgao1996 — CHANGES_REQUESTED** [review]
We should discuss this. I avoided using `move-deps` intentionally to dodge the feature unification problem caused by `move-deps`. 

`move-deps` depends on both `move-cli` and `move-stdlib`. `move-cli` depends on `move-stdlib` with feature "testing" enabled, forcing all downstream clients to also have this feature, resulting in some test-only natives being added to the production VM. 

I'm actually planning to put up a PR to make `framework` reference the individual Move crates, so I can fix the TODO I left in `aptos-gas/src/move_stdlib.rs`.

I do realize this increases the risk that someone forgets to update all the Move dependencies, but before we could take another look at our usage of features in the Move repo we'll likely need this for safety and correctness. How about I add some comments to `move-deps`'s Cargo.toml to document the issue and remind people that it's now not enough to just replace the hashes in this file?

**Comment by davidiw** [comment]
> We should discuss this. I avoided using `move-deps` intentionally to dodge the feature unification problem caused by `move-deps`.
> 
> `move-deps` depends on both `move-cli` and `move-stdlib`. `move-cli` depends on `move-stdlib` with feature "testing" enabled, forcing all downstream clients to also have this feature, resulting in some test-only natives being added to the production VM.
> 
> I'm actually planning to put up a PR to make `framework` reference the individual crates, so I can fix the TODO I left in `aptos-gas/src/move_stdlib.rs`.
> 
> I do realize this increases the risk that someone forgets to update all the Move dependencies, but before we could take another look at our usage of features in the Move repo we'll likely need this for safety and correctness. How about I add some comments to `move-deps`'s Cargo.toml to document the issue and remind people that it's now not enough to just replace the hashes in this file?

Sorry I missed that!

I do think that your approach though probably doesn't actually have the desired outcome. I imagine we're still building the natives in by default. If we really wanted to isolate, we'd have to make sure that every production import doesn't rely on the shared crate. Which I guess is what you're proposing. Feature unification in rust is a mess.

**Comment by vgao1996** [comment]
> I do think that your approach though probably doesn't actually have the desired outcome. I imagine we're still building the natives in by default.

Yeah I'm aware of this and this exactly what the TODO I left is for.

> If we really wanted to isolate, we'd have to make sure that every production import doesn't rely on the shared crate. Which I guess is what you're proposing. Feature unification in rust is a mess.

Indeed. We'll at least need to make `aptos-vm` use the individual crates. 

Let me study this a bit more and see if there are better solutions. If not, I guess we'll have to make the changes I suggested, at least for now.

**Comment by davidiw** [comment]
I believe if we move to the new feature resolver version 2 for cargo, we can safely integrate this with the desired outcome.

File: aptos-move/aptos-gas/Cargo.toml (hunk 1 above)
Comment (areshand) [reply aptos-move/aptos-gas/Cargo.toml:13]: just paste the link to the discussion https://aptos-org.slack.com/archives/C036X27DZNG/p1658952159877049

**Comment by vgao1996** [comment]
@davidiw we're already using resolver 2 and it wouldn't help in this case anyway: https://doc.rust-lang.org/cargo/reference/resolver.html#feature-resolver-version-2.

**Pull request closed without merging**